src/App.tsx

```diff
@@ -1,25 +1,101 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getTodos } from './api';
 
 const App: React.FC = () => {
-  const [
-    selectedUserId,
-    // setSelectedUserId,
-  ] = useState(0);
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+
+  const changeUser = (userId: number) => setSelectedUserId(userId);
+
+  const getCurrentTodos = () => {
+    getTodos()
+      .then((currentTodosFromServer) => {
+        setTodos(currentTodosFromServer);
+      });
+  };
+
+  useEffect(() => {
+    getCurrentTodos();
+  }, []);
+
+  const selectHandler = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    if (event.target.value === 'all') {
+      getCurrentTodos();
+    }
+
+    if (event.target.value === 'complited') {
+      getTodos()
+        .then((currentTodosFromServer) => {
+          setTodos(
+            currentTodosFromServer
+              .filter((todo: Todo) => todo.completed),
+          );
+        });
+    }
+
+    if (event.target.value === 'uncomplited') {
+      getTodos()
+        .then((currentTodosFromServer) => {
+          setTodos(
+            currentTodosFromServer
```

Comment:
and here

Reply:
Thank you for good marks!

But unfortunately, this way select work unproperly(I tried it yesterday, and today after your review).

The only way to write cleaner, that I see now - it's how you write previously - use switch-case and set state for switch property.

Anya, could you please tell me, is it necessary now - rewrite this way?